### DISCUSSION TITLE
setValue (for prefilling a form) is not working

### DISCUSSION THREAD

**Discussion by ranran2121** [react-hook-form/react-hook-form#7609] [Q&A]
Hi, I have an embarrassing question since in one component I used setValue for pre-filling a form, but the same procedure is not working in a similar component, where I just changed the name of the fields. 
This is how I did it: after I get the data from the query I put them in the setValue
````
useEffect(() => {
    if (loading === false) {
      const {
        id,
        deviceId,
        currentUserId,        
        active_status,
      } = data.getTransaction;

      setValue("active_status", active_status);
      setValue("currentUserId", currentUserId);
    }
  }, [loading]);
````
and, in the form: 
`````
 <form className='form'>
                    <div className='row mb-3'>
                    <label
                      htmlFor='currentUser'
                      className='col-sm-4 col-form-label'
                    >
                      Current User:
                    </label>
                    <div className='col-sm-8'>
                      <input
                        type='text'
                        className='form-control'
                        id='currentUser'
                        name='currentUserId'
                      />
                    </div>
                  </div>

                 <div className='row mb-3'>
                    <label className='col-sm-5 ' htmlFor='activeStatusSwitch'>
                      Active:
                    </label>
                    <div className='form-check form-switch col-sm-7'>
                      <input
                        className='form-check-input '
                        type='checkbox'
                        id='activeStatusSwitch'
                        name='active_status'
                      />
                    </div>
                  </div>
                </form>
`````
of course, the code is a bit more complex but the form is made of inputs (strings) and booleans (switches)
I checked in the development tools and the values are getting correctly to the component and to the Form (in the _formValues). Again, I did it exactly like in another component, the only difference is that the current form  is only for display purposes, that is the fields do not have to be modified (at the moment, but I do not know if they want to introduce this possibility)

**Comment by Moshyfawn** [comment]
Take a look at [this CSB](https://codesandbox.io/s/dynamic-radio-example-forked-eiqhj?file=/src/App.tsx) and tweak it to match your use case. I can't reproduce the issue

Also, consider using [`reset` form method](https://react-hook-form.com/api/useform/reset#main) instead of [setValue](https://react-hook-form.com/api/useform/setvalue) for setting your form's default values after the form has been already initialized

One thing I'll mention, but it might just the fact that you've omitted some code.. You still need to connect your input with your form values by [watching them](https://react-hook-form.com/api/useform/watch) or registering the input by using [`register`](https://react-hook-form.com/api/useform/register) or [`<Controller />`](https://react-hook-form.com/api/usecontroller/controller) / [`useController`](https://react-hook-form.com/api/usecontroller) to display your values. Keep in mind that since `register` provides the uncontrolled way of dealing with inputs, your user will be able to change the value if the input itself isn't disabled. In that case, consider using `field.value` from `<Controller />` / `useController`.

**Reply by ranran2121** [reply]
Hi, I believe I did not explain myself correctly.
1)I get the "default" values only after the query to the db is finished, so I cannot define defaultValues outside the component. I tried to assign the correctValues in the useEffect function but is is not working
2) so far the only way to have those value prefilled in the fields is to add the value attribute , for example
````
<form className='form'>
                    <div className='row mb-3'>
                    <label
                      htmlFor='currentUser'
                      className='col-sm-4 col-form-label'
                    >
                      Current User:
                    </label>
                    <div className='col-sm-8'>
                      <input
                        type='text'
                        className='form-control'
                        id='currentUser'
                        name='currentUserId'
                       value={data.currentUser}
                      />
                    </div>
                  </div>

                 <div className='row mb-3'>
                    <label className='col-sm-5 ' htmlFor='activeStatusSwitch'>
                      Active:
                    </label>
                    <div className='form-check form-switch col-sm-7'>
                      <input
                        className='form-check-input '
                        type='checkbox'
                        id='activeStatusSwitch'
                        name='active_status'
                       value={data.active_status}
                      />
                    </div>
                  </div>
                </form>
````

however the switch is not placed correctly in the "true" mode
is there a way to have the form without the register and 'on submit' option, since I do not want to submit the form (for the moment)?
PS: I do not know how to use Controller

**Reply by Peter-AMD** [reply]
I'm not quite sure but I think you're not leveraging enough of the react-hook-forms features by not using register or controller.

I think after your setValue you need to re-render again the form to reflect the changes. You can setState something after the setValues to test.

I highly recommend that you read the docs for a tutorial, rhf even have video tutorials in the docs page. RHF is such a great and powerful library if you fully use its features.

**Reply by Moshyfawn** [reply]
I agree with @Peter-AMD. To me it looks like you don't need RHF in this case at all, as it's a preview page for some information. Why setting `data.value` as the input's `value` doesn't work for you?
